### PR TITLE
ref(slack): Refactored SlackCommandsEndpoint and Link Identity View

### PR DESCRIPTION
**Pull request by iamrajjoshi**
https://www.notion.so/sentry/Slack-ff5b8ab4fd1e4760ab829438ce97ce15

Addresses `bot-6` task.

Added logging and metrics, not much else to clean up here

### CODE REVIEW

File: src/sentry/integrations/slack/views/link_team.py [hunk 1]
```diff
@@ -69,23 +78,36 @@ class SlackLinkTeamView(BaseView):
...
         if request.method not in ALLOWED_METHODS:
             return render_error_page(request, status=405, body_text="HTTP 405: Method not allowed")
 
         try:
-            params = unsign(signed_params)
-        except (SignatureExpired, BadSignature):
+            converted = unsign(signed_params)
+            params = TeamLinkRequest(**converted)
+        except (SignatureExpired, BadSignature) as e:
+            _logger.warning("handle.signature_error", exc_info=e)
```
Comment (cathteng) [inline src/sentry/integrations/slack/views/link_team.py:93]: should we specify that this is slack handling in the logger

File: src/sentry/integrations/slack/views/link_team.py [hunk 2]
```diff
@@ -69,23 +78,36 @@ class SlackLinkTeamView(BaseView):
...
-        integration = integration_service.get_integration(integration_id=params["integration_id"])
+        logger_params = asdict(params)
+        logger_params["user_id"] = request.user.id
+
+        integration = integration_service.get_integration(integration_id=params.integration_id)
         if integration is None:
-            raise Http404
+            _logger.info("integration.not_found", extra=logger_params)
+            metrics.incr(self._METRICS_FAILURE_KEY + ".get_integration", sample_rate=1.0)
+            raise render_error_page(
```
Comment (cathteng) [inline src/sentry/integrations/slack/views/link_team.py:108]: should this be `return`? do we want to know about Http404s?

**Review by cathteng — COMMENTED** [review]
are all of the metrics we're adding actionable? if it's user error like the team is already linked, do we need to be tracking that or is a logger sufficient?

File: src/sentry/integrations/slack/views/link_team.py [hunk 3]
```diff
@@ -35,6 +42,8 @@
 SUCCESS_LINKED_TITLE = "Team linked"
 SUCCESS_LINKED_MESSAGE = "The {slug} team will now receive issue alert{workflow_addon} notifications in the {channel_name} channel."
 
+_logger = logging.getLogger(__name__)
```
Comment (sentaur-athena) [inline src/sentry/integrations/slack/views/link_team.py:45]: Why _? Other loggers I see in the codebase are just `logger`

File: src/sentry/integrations/slack/views/link_team.py [hunk 4]
```diff
@@ -69,23 +78,36 @@ class SlackLinkTeamView(BaseView):
...
     @method_decorator(never_cache)
     def handle(self, request: Request, signed_params: str) -> HttpResponseBase:
         if request.method not in ALLOWED_METHODS:
             return render_error_page(request, status=405, body_text="HTTP 405: Method not allowed")
 
         try:
-            params = unsign(signed_params)
-        except (SignatureExpired, BadSignature):
+            converted = unsign(signed_params)
+            params = TeamLinkRequest(**converted)
```
Comment (sentaur-athena) [inline src/sentry/integrations/slack/views/link_team.py:91]: nit: should we rename this to team_link_request now that it's not params

File: src/sentry/integrations/slack/views/link_team.py [hunk 5]
```diff
@@ -173,6 +209,8 @@ def handle(self, request: Request, signed_params: str) -> HttpResponseBase:
 
         if not created:
             message = ALREADY_LINKED_MESSAGE.format(slug=team.slug)
+            _logger.info("team.already_linked", extra=logger_params)
```
Comment (sentaur-athena) [inline src/sentry/integrations/slack/views/link_team.py:212]: Is this a useful log?

File: src/sentry/integrations/slack/views/link_team.py [hunk 6]
```diff
@@ -202,17 +240,23 @@ def handle(self, request: Request, signed_params: str) -> HttpResponseBase:
...
         )
+
         integration_service.send_message(
             integration_id=integration.id,
             organization_id=team.organization_id,
             channel=channel_id,
             message=message,
         )
+
+        _logger.info("team.linked_success", extra=logger_params)
```
Comment (sentaur-athena) [inline src/sentry/integrations/slack/views/link_team.py:257]: Do we need both log as well as metric?

File: src/sentry/integrations/slack/views/link_team.py (hunk 3 above)
Comment (iamrajjoshi) [reply src/sentry/integrations/slack/views/link_team.py:45]: its a standard that Yash is trying to follow the convention of putting the `_` to emphasize that the logger is used within the file and shouldn't be exported. as part of the refactory, we have been removing references to common loggers in favor of this.

File: src/sentry/integrations/slack/views/link_team.py (hunk 1 above)
Comment (iamrajjoshi) [reply src/sentry/integrations/slack/views/link_team.py:93]: we should be able to get based on how we initialized the logger right `__name__`?

Comment (cathteng) [reply src/sentry/integrations/slack/views/link_team.py:93]: that's true

File: src/sentry/integrations/slack/views/link_team.py (hunk 2 above)
Comment (iamrajjoshi) [reply src/sentry/integrations/slack/views/link_team.py:108]: 1. yep should be return. 
2. we will still return a 404, but instead erroring and not returning anything to the user and leaving them with no context on what went wrong, we instead show them an error page. the logging and metric should give us info about the 404s anyway.

File: src/sentry/integrations/slack/views/link_team.py (hunk 6 above)
Comment (iamrajjoshi) [reply src/sentry/integrations/slack/views/link_team.py:257]: your right, ill remove it from here and a few other places i added this log.

File: src/sentry/integrations/slack/views/link_team.py (hunk 5 above)
Comment (iamrajjoshi) [reply src/sentry/integrations/slack/views/link_team.py]: good call, its unactionable, Ill remove it

**Review by RyanSkonnord — APPROVED** [review]

**Pull request merged**